### ISSUE TITLE
Switch block content based on request state?

### ISSUE THREAD

**Issue by eric-adstage** [DockYard/ember-async-button#78]
Hey guys,

I just discovered this plugin a few hours ago and have been playing around with it. How can I display different blocks within the async-button component based on the state of the request?

I see that the classBinding comes for free via ```textState```, but I would really like to be able to do something like (assume ```button``` passed into block):
```
{{#if button.isPending}}
    loader animation (svg)
{{else}}
    normal button text
{{/if}}
```

Whereas in the readme, simple text strings are used. How can I work with something more verbose/nuanced like the above? The code in #71 pretty much captures what I want to do, but it seems merging it was decided against, and I don't see any other real examples of what I'm trying to accomplish demonstrated elsewhere within the docs. Apologies if I've just overlooked something.

Thanks!

**Comment by manuel-reil** [comment]
+1

**Comment by bcardarella** [comment]
Until the API that I specified in #71 is implemented (which I believe needs to wait until the next version of HTMLBars) you should be able to use `ember-truth-helpers` to do what  you want:

```hbs
{{#if (eq button.textState "pending")}}
  loader animation (svg)
{{else}}
  normal button text
{{/if}}
```

I haven't tested this but I suspect this would work

**Comment by devinus** [comment]
I'm currently using `ember-truth-helpers` to do exactly this, but `is*` CPs would really clean this up.

**Comment by martndemus** [comment]
Implemented in #91, but needs a version release.
